packages/core/src/components/TextField/Mask.jsx

```diff
@@ -16,6 +16,15 @@ export class Mask extends React.PureComponent {
     };
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevState.debouncedOnBlurEvent && this.state.debouncedOnBlurEvent) {
+      this.field.props.onBlur(this.state.debouncedOnBlurEvent);
+
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ debouncedOnBlurEvent: null });
```

Comment:
Do we actually need `debouncedOnBlurEvent` to be in the state? Since we know it's already going to update because of the change in value, could we just attach it to the component instance? (`this.debouncedOnBlurEvent = `)

Reply:
Hmm, I don't have strong feelings on that so am happy to change. I'm not sure what the performance impacts are of either option tbh.

Reply:
There's a chance `state.value` could be updating for a different reason, but that's why we still check whether `this.debouncedOnBlurEvent` exists before using it/clearing it.

Reply:
I think you're right though, since changing the state would require checking if the component needs re-rendered, whereas I don't think that's the case if it's attached to the component instance.